### ISSUE TITLE
Add check for BQ compatible schema-changes to PR review

### ISSUE THREAD

**Issue by fbertsch** [mozilla-services/mozilla-pipeline-schemas#316]
This comes from the [GCP schema handling doc.](https://docs.google.com/document/d/1s1Zf-RVD9qxu5goIOjMBTtjyzKQbNsoAotYpR8T6QQg/edit#heading=h.5tnpqz2an9ah)

We will need to initially start minimizing schema-incompatible changes for BQ. This only means that we can't change field types and can't remove fields. The former has historically (AFAIK) not been a common change, and the latter can be easily avoided by simply keeping the fields.

**Comment by jklukas** [comment]
Note on implementation: this test will likely need to pull the pre-PR state of any changed schemas via some `git` commands, then do some amount of parsing of the two versions of the schema to detect the conditions we want to avoid.

**Comment by fbertsch** [comment]
Those are good points @jklukas! I meant for this to just be added to the PR review checklist. Mark and I were discussing a longer-term CI test for schema changes, with a short-term improvement being akin to the one you mentioned, should we move that to a separate issue?

**Comment by jklukas** [comment]
> should we move that to a separate issue

Indeed. I interpreted this as being about CI tests. Apologies for missing the scope. I agree that adding to a checklist is a good first step that can be done immediately.

**Comment by fbertsch** [comment]
Okay great. Updated the title of this issue to make it more clear, and let's move CI discussion to #317.
